cmd/codegen/generator/go/generator.go

```diff
@@ -64,6 +64,48 @@ func (g *GoGenerator) Generate(ctx context.Context, schema *introspection.Schema
 
 	mfs := memfs.New()
 
+	if g.Config.ClientOnly {
```

Comment:
Can we split this into an entirely separate function? It doesn't really share *any* logic with the rest of this, we could have `GenerateClient` here, as another method in the interface. This then matches the `generateClient` in the engine SDK interface.